jsonnet/openebs-mixin/README.md

```diff
@@ -0,0 +1,62 @@
+# openebs mixin
+
+A set of Grafana dashboards and Prometheus alerts for openebs.
+
+## Generate dashboards and rules
+
+You can manually generate the alerts, dashboards and rules files. 
+
+#### Steps:
+
+1. Install `jsonnet` and `jb`
+	```
+	$ go get github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb
+	$ brew install jsonnet
+	```
+
+2. Then, grab the mixin and its dependencies:
+
+	```
+	$ git clone https://github.com/openebs/monitoring
+	$ cd monitoring/jsonnet/openebs-mixin
+	$ jb install
+	```
+
+3. Update the dashboards and rules directory path in `build.sh`
+	```
+	dashboardsDirPath = ''
+	rulesDirPath = ''
+	```
+
+4. Finally, build the mixin:
+
+	```
+	$ make generate
+	```
+
+The  files in `rulesDirPath` are the alert rules defined for openebs which has to be used to create prometheusRule yaml then need to passed to your Prometheus server, and the files in `dashboardsDirPath` need to be imported into you Grafana server. 
+
+## Configuration
+
+The available fields and their default values can be seen in `config.libsonnet`. 
+
+```
+{
+  // openebs-mixin configurations
+  _config+:: {
+    // Dashboards configuration. If set, then dashboards json will be generated.
+    dashboards: {
+      cStor: true,
+      Jiva: true,
+      LocalPV: true,
+    },
+    // AlertRules configuration. If set, then rules json will be generated.
+    alertRules: {
+      cStor: true,
+      Jiva: true,
+      lvmLocalPV: false,
+      volume: true,
```

Comment:
Since this category `volume` contains common rules for all types of CAS engines, will it be better if we remove the switch for this and include the rules whenever any of the above CAS types are set as true?
The contention is, why would anyone want to enable cStor=true, but volume=false?

Reply:
Should we specify npd as well here?

Reply:
Makes sense.
We can do an OR operation to check whether any of the CAS types are set as true.
 `volume: $._config.alertRules.cStor || $._config.alertRules.Jiva || $._config.alertRules.lvmLocalPV`

Reply:
Ok, make it a non-configurable variable then.

Reply:
This configuration section is not in sync with the actual configuration.